application/view/common/block-layout/browse-preview.phtml

```diff
@@ -16,9 +16,15 @@ foreach ($this->resources as $resource):
     $body = $bodyTerm ? $resource->value($bodyTerm) : $resource->displayDescription();
 ?>
     <li class="<?php echo $this->resourceType; ?> resource">
-        <?php echo $resource->linkRaw($this->thumbnail($resource, 'medium', ['title' => $heading])); ?>
-        <h4><?php echo $resource->link($heading); ?></h4>
-        <?php if ($body): ?>
+        <?php
+        if (in_array('thumbnail', $this->components)):
+            echo $resource->linkRaw($this->thumbnail($resource, 'medium', ['title' => $heading]));
+        endif;?>
+        <h4><?php
```

Comment:
This `if` should be moved: as is it will output an empty `h4` element if the heading is disabled, when the `h4` should be removed completely in that case.